### ISSUE TITLE
Distinction between Vimperator and Muttator

### ISSUE THREAD

**Issue by SammysHP** [vimperator/vimperator-labs#617]
I just wanted to implement a new feature that is Firefox-only and wondered where I should start. Most code is in `common/content` which is shared between Vimperator and Muttator.

Actually lots of code in `common` is not meant for Thunderbird/Muttator. Even worse: There are version checks in it that compare the version number (e.g. [common/content/sanitizer.js#L111](https://github.com/vimperator/vimperator-labs/blob/8f0025eef320846d210ee97b02a38973f7cbb13a/common/content/sanitizer.js#L111)) without any distinction between Firefox and Thunderbird.

What should we do? Firefox and Thunderbird divert even more with time. I wonder if Vimperator can exists in the future (e.g. multi process) while keep sharing code with Thunderbird.

**Comment by maxauthority** [comment]
Yeah, that's a very interesting question. I personally haven't used
Muttator for quite a while, so I am very impressed if it actually still
works. For now the stance was to not deliberatly break Muttator, but even
with some small quirks it's better not to abandon Muttator fully.

If Vimperator was rewritten as an WebExtension, then I think it's quite
clear that Muttator would need to be dropped (did they even announce
WebExtensions for Thunderbird?). On the other hand, if your new feature
(what is it?, just being curious ;)) warrants a seperate file, then putting
that in vimperator/content/ instead of common/content/ would probably be
better.

--
Martin

**Comment by SammysHP** [comment]
> what is it?, just being curious ;)

It's #619 which I already implemented in #620. I just added the new option to `vimperator/content/config.js`.

**Comment by petelewis** [comment]
Please don't abandon muttator :-)

I only wish it had proper Vim integration. I was wondering if that might even be feasible now with neovim...

**Comment by stepshal** [comment]
@maxauthority Make new muttator release please.

**Comment by maxauthority** [comment]
Yeah, sounds like a good idea. I'll also give @gkatsev owner rights on addons.mozilla.org so he can also make releases as he likes :) I just need to first find out, how to login to addons.mozilla.org first, as my usual username/password combo doesn't work anymore but it asks for a "Firefox account" with an email address (and mine isn't found). Has anybody had the same issue and do I really need to create (and associate) a new Firefox account?

**Comment by SammysHP** [comment]
I got a mail some weeks ago to migrate my account. Here's an help article:

https://support.mozilla.org/en-US/kb/access-your-add-ons-firefox-accounts

**Comment by maxauthority** [comment]
Oh, must have missed that! Thanks a lot! Will do it tomorrow or so, and then add (at least) @gkatsev to the "people with power" on AMo for Muttator, so future updates should be much more reliable if it is not depending on a single person.

**Comment by maxauthority** [comment]
Ok, I tried releasing Muttator 1.4, but it didn't pass the review :( They seem to get more strict about the usage of eval. Wondering though, why Vimperator 3.15 was accepted with the same usage. Maybe we just had "bad luck", but still not sure, how to continue best:
----------------------
Hello,

Your add-on, Muttator 1.4, has been reviewed and did not meet the criteria for being hosted in our gallery.

Reviewer:
Leszek Ż****

Comments:
This version didn't pass review because of the following problems:

1. We generally don't accept using the 'eval' function. There are many reasons not to use 'eval', and there are alternatives available. You can read more about it here: https://developer.mozilla.org/en/XUL_School/Appendix_C:_Avoid_using_eval_in_Add-ons

This version of your add-on has been disabled. You may re-request review by addressing the reviewer's comments and uploading a new version at https://addons.mozilla.org/en-US/developers/addon/muttator/versions


If you want to respond to this review, or have any questions about it, please reply to this email or join #amo-editors on irc.mozilla.org.

To learn more about the review process, please visit https://developer.mozilla.org/en-US/Add-ons/AMO/Policy/Reviews
--
Mozilla Add-ons Team
https://addons.mozilla.org
